src/main/java/jacchm/footballapp/service/CompetitionService.java

```diff
@@ -0,0 +1,95 @@
+package jacchm.footballapp.service;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import jacchm.footballapp.mapping.inputs.CompetitionsInput;
+import jacchm.footballapp.mapping.dto.CompetitionDTO;
+import jacchm.footballapp.model.entity.Competition;
+import jacchm.footballapp.mapping.mapper.CompetitionMapper;
+import jacchm.footballapp.repository.CompetitionRepository;
+import jacchm.footballapp.util.JsonUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class CompetitionService {
+
+    private static final Logger logger = LoggerFactory.getLogger(CompetitionService.class);
+    private File file = new File("src\\main\\" +
+            "resources\\dataFromExternalApi\\AllCompetitions.json");
+
+    private CompetitionRepository competitionRepository;
+    private ExternalFootballAPIService externalFootballAPIService;
+
+    @Autowired
+    public void setCompetitionRepository(CompetitionRepository competitionRepository) {
+        this.competitionRepository = competitionRepository;
+    }
+
+    @Autowired
+    public void setExternalFootballAPIService(ExternalFootballAPIService externalFootballAPIService) {
+        this.externalFootballAPIService = externalFootballAPIService;
+    }
+
+
+    public String deleteAll() {
+        competitionRepository.deleteAll();
+        return "Delete has been completed.";
+    }
+
+    public List<CompetitionDTO> getAll() {
+        List<CompetitionDTO> allCompetitionsDTO = new ArrayList<>();
+        for (Competition competition : competitionRepository.findAll()) {
+            allCompetitionsDTO.add(CompetitionMapper.INSTANCE.competitionToCompetitionDTO(competition));
+        }
+
+        return allCompetitionsDTO;
+    }
+
+    public CompetitionDTO getById(Integer id) {
+        Competition toBeReturned = competitionRepository.findById(id).orElse(new Competition());
+        return CompetitionMapper.INSTANCE.competitionToCompetitionDTO(toBeReturned);
+    }
+
+
+    // TODO: scheduled or done manually by admin
+    public String updateCompetitions() {
+        String jsonCompetitionInput;
+        CompetitionsInput competitionsInputDTO;
+        JsonNode node;
+
+        try {
+            jsonCompetitionInput = externalFootballAPIService.getCompetitions();
+            node = JsonUtil.parse(jsonCompetitionInput);
+            competitionsInputDTO = JsonUtil.fromJson(node, CompetitionsInput.class);
+
+            logger.info("Connection with external API has been established. Competitions have been achieved.");
+
+        } catch (IOException e1) {
+            // In case that external football API doesn't respond
+            logger.error("External Football API hasn't responded. Program will try to achieve the data from file.");
+            try {

```

Comment:
try catch w środku catch nie wygląda za dobrze, jak chcesz tylko zalogować że coś się nie udało to ok, ale zamiast dawać takie zagnieżdżenie zrób sobie try za blokiem catch a nie wewnątrz